src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,28 +561,63 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
+                While CanGet(Here)
+                    ch = Peek(Here)
+                    If IsWhitespace(ch) Then
+                        Here += 1
+                    Else
+                        Exit While
+                    End If
+                End While
+                If CanGet(Here) AndAlso IsSingleQuote(Peek(Here)) Then
+                    ' If you get here you have a Line Continuation with have 0 or more spaces followed by a  comment
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
+                    Dim comment As SyntaxTrivia = ScanComment()
+                    If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                        comment = DirectCast(comment.SetDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_CommentsAfterLineContinuationNotAvailable1,
+                                                                                                    New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))}), SyntaxTrivia)
+                    End If
+                    tList.Add(comment)
+                    ch = Peek()
+                    atNewLine = IsNewLine(ch)
                 Else
-                    Exit While
+                    ' We have a Line Continuation without comment but have 0 or more spaces after _, so process as V15.5
+                    If Not CanGet(Here) Then
+                        Return False
+                    End If
+                    atNewLine = IsNewLine(Peek(Here))
+                    If Not atNewLine Then
+                        Return False
+                    End If
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    ' Leading Whitespace on line after _
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
                 End If
-            End While
-
-            ' Line continuation is valid at the end of the
-            ' line or at the end of file only.
-            Dim atNewLine = IsNewLine(ch)
-            If Not atNewLine AndAlso CanGet(Here) Then
-                Return False
-            End If
-
-            tList.Add(MakeLineContinuationTrivia(GetText(1)))
-            If Here > 1 Then
-                tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+            Else
+                ' We don't have a space or ' but we might have an EOF after _ and that is not an error
+                ' This case if different then above because we don't want to skip whitespace after _
+                If CanGet(Here) Then
```

Comment:
>CanGet(Here) [](start = 19, length = 12)

Can we return ```False``` if this condition is false, or can this function succeed even in this situation?

Reply:
I guess it can succeed.

---
In reply to: [213399884](https://github.com/dotnet/roslyn/pull/28327#discussion_r213399884) [](ancestors = 213399884)

Reply:
If you return false there things break when typing new code and type an _ at the end of the file. There is a test to show this and I added a comment to explain why.